src/Microsoft.ServiceFabric.Actors/Diagnostics/ActorMethodInfo.cs

```diff
@@ -0,0 +1,24 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System.Reflection;
+
+namespace Microsoft.ServiceFabric.Actors.Diagnostics
+{
+    internal class ActorMethodInfo
+    {
+        internal ActorMethodInfo() { }
+
+        internal ActorMethodInfo(MethodInfo methodInfo)
+        {
+            MethodName = string.Concat(methodInfo.DeclaringType.Name, ".", methodInfo.Name);
+            MethodSignature = methodInfo.ToString();
+        }
+
+        internal string MethodName { get; set; }
```

Comment:
Can we define these as read-only fields instead of read-write properties? The design guideline recommending properties shouldn't apply to internal APIs.